### PR TITLE
Rename flip() to reverse()

### PR DESCRIPTION
**Pull request by cbracken**
Deprecate flip() for one release, then remove.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+#### Unreleased
+   * Strings: Rename `flip` to `reverse`. `flip` is deprecated and will be
+     removed in one release.
```
Comment (yjbanov) [inline CHANGELOG.md]: "one"? Did you mean "1.0"? Or "a future"?

Comment (yjbanov) [inline CHANGELOG.md]: Also, this might technically be a breaking change since we could create conflicts with `reverse`s from other libraries.

Comment (cbracken) [inline CHANGELOG.md]: I mean in the current version++. Reworded to 'the next release'. 

In theory every addition of a new top-level identifier is a potentially breaking change. Not sure if it's worth calling them out as such though. Added a bit more detail to the changelog.

**Pull request merged**